xarray/tests/test_coding_strings.py

```diff
@@ -29,6 +36,33 @@ def test_vlen_dtype() -> None:
     assert strings.check_vlen_dtype(np.dtype(object)) is None
 
 
+@pytest.mark.parametrize("numpy_str_type", (np.str_, np.bytes_))
+def test_numpy_subclass_handling(numpy_str_type) -> None:
+    with pytest.raises(TypeError, match="unsupported type for vlen_dtype"):
+        strings.create_vlen_dtype(numpy_str_type)
+
+
+@requires_netCDF4
+@pytest.mark.parametrize("str_type", (str, np.str_))
+def test_write_file_from_np_str(str_type) -> None:
+    # should be moved elsewhere probably
```

Comment:
Please move this to test_backends.py, and include a comment linking back to the GitHub issue (for context).

Reply:
👍